src/Types/DateTimeType.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Types;
+
+use DateTime;
+use DateTimeInterface;
+use Exception;
+use GraphQL\Language\AST\Node;
+use Overblog\GraphQLBundle\Annotation as GQL;
+
+#[GQL\Scalar('DateTime')]
+class DateTimeType
+{
+    public static function serialize(DateTime $value): string
+    {
+        return $value->format(DateTimeInterface::RFC3339);
+    }
+
+    /**
+     * @throws Exception
+     */
+    public static function parseValue(string $value): DateTime
+    {
+        return new DateTime($value);
+    }
+
+    /**
+     * @throws Exception
+     */
+    public static function parseLiteral(Node $valueNode): DateTime
+    {
+        $value = $valueNode->value; // @phpstan-ignore-line
```

Comment:
This is specified by [`overblog/GraphQLBundle`](https://github.com/overblog/GraphQLBundle/blob/master/docs/definitions/type-system/scalars.md#custom-scalar).

Reply:
Oh wow.  
I believe it _should_ be an `IntValueNode`, so if the method signature does not have to be exactly this, we can do
```php
    public static function parseLiteral(IntValueNode $valueNode): \DateTime
    {
        return new \DateTime($valueNode->value);
    }
```
otherwise we need to add a doc comment for it
```php
    /**
     * @param IntValueNode $valueNode
     * @throws Exception ...
     */
```

Reply:
It should actually be a string, as the DateTime constructor expects a string.

Potentially, it can be anything [possible](https://github.com/webonyx/graphql-php/blob/master/src/Language/AST/Node.php#L9). By specifying a special type, you are suggesting a false assumption. I don't think you can enforce a type here if the underlying lib doesn't provide for it.

Reply:
oh my bad - maybe we then just check if the value exists?
```php
        if (!property_exists($valueNode, 'value')) {
            throw new \InvalidArgumentException(
                'expected node with string value, got ' . get_class($valueNode)
            );
        }
```